servers/rendering_server.cpp

```diff
@@ -31,6 +31,13 @@
 #include "rendering_server.h"
 #include "rendering_server.compat.inc"
 
+#if defined(VULKAN_ENABLED)
+#include "drivers/vulkan/rendering_context_driver_vulkan.h"
+#endif
+#if defined(METAL_ENABLED)
+#include "drivers/metal/rendering_context_driver_metal.h"
+#endif
```

Comment:
This is a pre-existing issue, so maybe not blocking, but these includes break code encapsulation. The generic / abstract `servers` code shouldn't need to access driver code, and instead have virtual methods that drivers can implement.

Reply:
I'm trying to think of the best way to do this. Does it make sense to have this in the DisplayServer class?
Since the RenderServer doesn't have API.
I don't know the subtlety between the two classes. Does it even make sense for the RenderServer to have `RenderingServer::create_local_rendering_device()` ?